osu.Game/Overlays/UserButton.cs

```diff
@@ -0,0 +1,150 @@
+﻿//Copyright (c) 2007-2016 ppy Pty Ltd <contact@ppy.sh>.
+//Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using System;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Input;
+using osu.Game.Graphics;
+using OpenTK;
+using OpenTK.Graphics;
+using osu.Framework;
+using osu.Framework.Graphics.Primitives;
+
+namespace osu.Game.Overlays
+{
+    class UserButton : Container
+    {
+        public const float WIDTH = 60;
+
+        public Sprite Image
+        {
+            get { return DrawableAvatar; }
+            set
+            {
+                DrawableAvatar.Texture = value.Texture;
+            }
+        }
+
+        public string Text
+        {
+            get { return DrawableText.Text; }
+            set
+            {
+                DrawableText.Text = value;
+            }
+        }
+
+        public string TooltipMain
+        {
+            get { return tooltip1.Text; }
+            set
+            {
+                tooltip1.Text = value;
+            }
+        }
+
+        public string TooltipSub
+        {
+            get { return tooltip2.Text; }
+            set
+            {
+                tooltip2.Text = value;
+            }
+        }
+
+        public Action Action;
+        protected Avatar DrawableAvatar;
+        protected SpriteText DrawableText;
+        protected Box HoverBackground;
+        private FlowContainer tooltipContainer;
+        private SpriteText tooltip1;

```

Comment:
these names aren't very descriptive, the Main/Sub distinction you used in the public properties would be fine if you reused it:

```
private tooltipSub;
public TooltipSub;
```